packages/core/src/types.ts

```diff
@@ -740,6 +740,119 @@ export type DelayConfig<TContext, TEvent extends EventObject> =
   | number
   | DelayExpr<TContext, TEvent>;
 
+export type RequiredTypegenMachineOptions<
+  TContext,
+  TEvent extends EventObject,
+  TMeta extends TypegenMeta
+> = Required<
+  Pick<
+    TypegenMachineOptions<TContext, TEvent, TMeta, 1>,
+    GetRequiredTypegenOptionKeys<TMeta>
+  >
+>;
```

Comment:
As far as I can tell this `Required` is redundant because all the keys of `TypegenMachineOptions` are already required
```suggestion
> = Pick<
  TypegenMachineOptions<TContext, TEvent, TMeta, 1>,
  GetRequiredTypegenOptionKeys<TMeta>
>;
```